File: BACK-END/KuentecoApp/src/main/java/org/kuenteco/backend/dto/logic/transaction/kuenteco/UpdateTransactionDTO.java
package org.kuenteco.backend.dto.logic.transaction.kuenteco;

import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kuenteco.backend.entity.extra.DescriptionTransaction;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UpdateTransactionDTO {
    @NotNull private Integer id;
    @NotNull private Integer categoryId;
    @NotNull private Integer budgetId;
    @NotNull private Integer debtId;
    @NotNull private DescriptionTransaction description;
    @NotNull private BigDecimal amount;
}

File: FRONT-END/lib/dto/transaction/kuenteco/update_transaction_dto.dart
import '../../extra/description_transaction_extra.dart';

class UpdateTransactionDTO {
  final int id;
  final int categoryId;
  final int budgetId;
  final int debtId;
  final String? name;
  final DescriptionTransaction description;
  final double amount;

  UpdateTransactionDTO({
    required this.id,
    required this.categoryId,
    required this.budgetId,
    required this.debtId,
    this.name,
    required this.description,
    required this.amount,
  });

  factory UpdateTransactionDTO.fromJson(Map<String, dynamic> json) {
    return UpdateTransactionDTO(
      id: json['id'],
      categoryId: json['categoryId'],
      budgetId: json['budgetId'],
      debtId: json['debtId'],
      name: json['name'],
      description: DescriptionTransaction.fromJson(json['description']),
      amount: (json['amount'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'categoryId': categoryId,
      'budgetId': budgetId,
      'debtId': debtId,
      'name': name,
      'description': description.toJson(),
      'amount': amount,
    };
  }
}
